src/v6/signing.ts

```diff
@@ -0,0 +1,253 @@
+import { _TypedDataEncoder } from "ethers/lib/utils"
+import { keccak_256 } from "@noble/hashes/sha3"
+import { hexToBytes, utf8ToBytes } from "@noble/hashes/utils"
+import { AllowFill, GenericSwapData, LimitOrder, RFQOffer, RFQTx } from "./types"
+
+import {
+    EIP712DomainOptions,
+    EIP712Types,
+    SigningHelper as BaseSigningHelper,
+    SigningOptions,
+} from "../signing"
+
+export class SigningHelper extends BaseSigningHelper {
+    public constructor() {
+        super({
+            name: "Tokenlon",
+            version: "v6",
+        })
+    }
+
+    private toBuffer = (data: string, encoding: "utf8" | "hex" = "hex"): Uint8Array => {
+        if (encoding === "hex") {
+            if (data.startsWith("0x")) {
+                return hexToBytes(data.substring(2))
+            }
+            return hexToBytes(data)
+        }
+        return utf8ToBytes(data)
+    }
+
+    private keccak256 = (data: string | Uint8Array, encoding?: "utf8" | "hex"): String => {
+        if (typeof data === "string" && encoding === "utf8") {
+            return "0x" + Buffer.from(keccak_256(this.toBuffer(data, encoding))).toString("hex")
+        }
+        return "0x" + Buffer.from(keccak_256(data)).toString("hex")
+    }
```

Comment:
I think `ethers.js` have `keccack256` util, did you try it before?

https://docs.ethers.org/v5/api/utils/hashing/#utils-keccak256

Reply:
Refined `getXXXEIP712Typehash` for each type and used keccak from ethers